controlpanel/frontend/static/javascripts/modules/tool-status.js

```diff
@@ -38,10 +38,13 @@ moj.Modules.toolStatus = {
           break;
         case 'READY':
         case 'IDLED':
-          this.showActions(listener, ['open', 'restart', 'remove']);
+          this.showActions(listener, ['open', 'restart', 'upgrade', 'remove']);
+          break;
+        case 'UPGRADED':
+          this.showActions(listener, ['open']);
```

Comment:
Why doesn't an upgraded tool have `restart`? (and `remove` - not that it exists yet)

Reply:
no `restart`: I imagine no point to restart something which was just upgraded - but I can add if you disagree
no `remove`: because it doesn't exist - although I can add if you think it's good 
 to leave for the future

Speaking of `restart`: does it make sense when something is already idled? I was tempted to remove but then I left as it was a bit out of scope and I thought there may have been a reason for it.

Reply:
They may want to restart if the tool fails to unidle.

Reply:
Wouldn't that be a separate state tho? It would be probably be "unknown" or "failed" no? (maybe I miss something).